idmapping-rest/src/main/java/org/uniprot/api/idmapping/service/PIRResponseConverter.java

```diff
@@ -101,46 +101,59 @@ public IdMappingResult convertToIDMappings(
             } else {
-                String fromValue = getFromValue(mappedId, rowParts[0]);
-                Arrays.stream(rowParts[1].split(";"))
-                        // filter based on valid to
-                        .filter(toValue -> isValidIdPattern(request.getTo(), toValue))
-                        .map(
-                                toValue ->
-                                        IdMappingStringPair.builder()
-                                                .from(fromValue)
-                                                .to(toValue)
-                                                .build())
-                        .forEach(builder::mappedId);
+                for (String fromValue : fromValues) {
+                    Arrays.stream(rowParts[1].split(";"))
+                            // filter based on valid to
+                            .filter(toValue -> isValidIdPattern(request.getTo(), toValue))
+                            .map(
+                                    toValue ->
+                                            IdMappingStringPair.builder()
+                                                    .from(fromValue)
+                                                    .to(toValue)
+                                                    .build())
+                            .forEach(builder::mappedId);
+                }
             }
         }
     }
 
-    private String getFromValue(Map<String, String> mappedIds, String fromValue) {
-        return mappedIds.getOrDefault(fromValue, fromValue);
+    private Set<String> getFromValue(Map<String, Set<String>> mappedIds, String fromValue) {
+        return mappedIds.getOrDefault(fromValue, Set.of(fromValue));
     }
 
-    Map<String, String> getMappedRequestIds(IdMappingJobRequest request) {
-        Map<String, String> mappedIds = new HashMap<>();
+    Map<String, Set<String>> getMappedRequestIds(IdMappingJobRequest request) {
+        Map<String, Set<String>> mappedIds = new HashMap<>();
         if (ACC_ID_STR.equals(request.getFrom())
                 && (request.getIds().contains("[") || request.getIds().contains("."))) {
             String ids = String.join(",", request.getIds());
             mappedIds =
                     Arrays.stream(ids.split(","))
                             .map(this::getMappedId)
                             .filter(Objects::nonNull)
-                            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+                            .collect(
+                                    Collectors.groupingBy(
+                                            Map.Entry::getKey,
+                                            Collectors.mapping(
+                                                    Map.Entry::getValue,
+                                                    Collectors.toCollection(LinkedHashSet::new))));
+            // remove more than one versions from values
```

Comment:
Can you please explain me why we are removing the version for just first value?

Reply:
if we have more than one version of the same accession, we return just main entry with mapping.
e.g. P05067.304 -> P05067
If we don't remove version, we will have duplicate mapping and it will be different that current implementation